src/orders/OrderProcessor.sol

```diff
@@ -18,6 +18,7 @@ import {IdShare} from "../IdShare.sol";
 import {FeeLib} from "../common/FeeLib.sol";
 import {IForwarder} from "../forwarder/IForwarder.sol";
 import {IFeeSchedule} from "./IFeeSchedule.sol";
+import {IVault} from "../IVault.sol";
```

Comment:
Remove Vault from OrderProcessor. Contracts/modules should be isolated. If needed, write a separate router to bundle specific calls together. This allows for flexibility over time.